pages/admin.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+import { withRouter } from "next/router";
+import jsyaml from "js-yaml";
+import PT from "prop-types";
+
+import "../styles/admin.scss";
+
+import Layout from "../components/Layout";
+import Drawer from "../components/Drawer";
+import RenderLimberSettings from "../components/RenderLimberSettings";
+
+import RequestService from "../util/requestService";
+
+function AdminPage({ router, limberSettings }) {
+
+	// PARSE "limber/settings.yml" content as JS Object and PUT to the State BEFORE on page render
+	useEffect(() => {
+		_parseSettings(limberSettings);
+	}, []);
+
+	// DEFINE State
+	const [settings, setSettings] = useState({});
+
+	// PARSE "limber/settings.yml" content as JS Object and PUT to the State
+	async function _parseSettings(settings) {
+		const settingsFileContent = settings.content;
+		const fromBase64ToString = window.atob(settingsFileContent);
+		setSettings(jsyaml.load(fromBase64ToString));
+	}
+
+	return (
+		<>
+			<Layout>
+				<Drawer activePath={router.asPath} limberData={settings} />
+				{
+					router.asPath === "/admin" && (
+						<>
+							<h1 className="o-title">Welcome to the Admin page of the Git CMS</h1>
+						</>
+					)
+				}
+
+				{
+					router.asPath === "/admin?page=settings" && (
+						<>
+							<RenderLimberSettings settings={settings} />
```

Comment:
It depends on the preferences, I described above.

I was thinking about handling everything as Single-page (e.g. having all the logic done at `/admin` page). But seems that it was a bad idea because, in the end, it will be much harder to maintain and expand.

So, yeah, I agree that it is a better way.